File: mid-term-exam/backend/src/items_inventory/domain/model/Item.ts
export default class Item {
  private id: number;
  private name: string;
  private vendor: string;
  private rating: number;
  private imagePath: string;

  constructor(
    id: number,
    name: string,
    vendor: string,
    rating: number,
    imagePath: string
  ) {
    this.id = id;
    this.name = name;
    this.vendor = vendor;
    this.rating = rating;
    this.imagePath = imagePath;
  }

  getId(): number {
    return this.id;
  }

  getName(): string {
    return this.name;
  }

  getVendor(): string {
    return this.vendor;
  }

  getRating(): number {
    return this.rating;
  }

  getImagePath(): string {
    return this.imagePath;
  }
}

File: mid-term-exam/app/items_list/lib/model/item.dart
class Item {
  final int id;
  final String name;
  final String vendor;
  final double rating;
  final String imagePath;

  Item(
      {required this.id,
      required this.name,
      required this.vendor,
      required this.rating,
      required this.imagePath});

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
        id: json['id'],
        name: json['name'],
        vendor: json['vendor'],
        rating: json['rating'],
        imagePath: json['imagePath']);
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'vendor': vendor,
      'rating': rating,
      'imagePath': imagePath
    };
  }
}
